metadata-service/servlet/src/main/java/com/datahub/gms/servlet/Config.java

```diff
@@ -136,11 +140,17 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
       }
     }
 
-    // Serialize cached configuration
+    // Serialize cached configuration with user context
     configLock.readLock().lock();
     try {
       resp.setContentType("application/json");
-      String json = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(cachedConfig);
+
+      // Create a copy of cached config and add user type for progressive disclosure
+      Map<String, Object> responseConfig = new HashMap<>(cachedConfig);
+      responseConfig.put("userType", determineUserType());
```

Comment:
DSN information (that I should include for authenticated users) does not exist in this repo, I could add a placeholder here (as a section in application.yaml) or it can be added in the fork. I'm not sure what is the correct way. My current instinct is to remove this extra logging and merge it and add the DSN as a follow up PR either here or in the fork

Reply:
I think follow up PR once the requirements are fleshed out makes more sense than placeholder to potentially refactor